superset-frontend/src/SqlLab/components/HighlightedSql.jsx

```diff
@@ -41,77 +41,66 @@ const propTypes = {
   shrink: PropTypes.bool,
 };
 
-class HighlightedSql extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      modalBody: null,
-    };
-  }
+function HighlightedSql({ sql, rawSql, maxWidth, maxLines, shrink }) {
+  const [modalBody, setModalBody] = useState(null);
 
-  shrinkSql() {
-    const ssql = this.props.sql || '';
+  const shrinkSql = () => {
+    const ssql = sql || '';
     let lines = ssql.split('\n');
-    if (lines.length >= this.props.maxLines) {
-      lines = lines.slice(0, this.props.maxLines);
+    if (lines.length >= maxLines) {
+      lines = lines.slice(0, maxLines);
       lines.push('{...}');
     }
     return lines
       .map(line => {
-        if (line.length > this.props.maxWidth) {
-          return `${line.slice(0, this.props.maxWidth)}{...}`;
+        if (line.length > maxWidth) {
+          return `${line.slice(0, maxWidth)}{...}`;
         }
         return line;
       })
       .join('\n');
-  }
+  };
 
-  triggerNode() {
-    const shownSql = this.props.shrink
-      ? this.shrinkSql(this.props.sql)
-      : this.props.sql;
+  const triggerNode = () => {
+    const shownSql = shrink ? shrinkSql(sql) : sql;
```

Comment:
looks like you don't need to pass the arg here.

Reply:
(I see that was how it was earlier, so just a bit of proactive cleanup since we're here)

Reply:
I run into a bunch of syntax errors when I reformat this to be 

```
const triggerNode = (
   **code**
)
```

The biggest thing, I think, is that there is another const defined within it? Could that have something to do with it. I did some reading, and looked at a fair amount of previous code, and that was the only thing that this has that others don't.